src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs

```diff
@@ -20,6 +20,9 @@ public abstract class AggregateType : FormulaType
         /// </summary>
         public virtual string UserVisibleTypeName => null; 
 
+        // Internal property, allows Canvas to control whether a Union operation is permitted on a derived record type
+        internal virtual bool AllowsUnion => true;
```

Comment:
May want to make this public in the future. Applies to cases like
`If(x, DerivedRecord1, DerivedRecord2)` and controls whether that will type check or not. We may want to allow derived record types beyond controls to restrict this.